concrete-core/src/backends/core/implementation/engines/ggsw_ciphertext_encryption.rs

```diff
@@ -0,0 +1,178 @@
+};
+use crate::specification::entities::GlweSecretKeyEntity;
+
+/// # Description:
+/// Implementation of [`GgswCiphertextEncryptionEngine`] for [`CoreEngine`] that operates on 32 bits
+/// integers.
+impl GgswCiphertextEncryptionEngine<GlweSecretKey32, Plaintext32, GgswCiphertext32> for CoreEngine {
+    /// # Example:
+    /// ```
+    /// use concrete_commons::dispersion::Variance;
+    /// use concrete_commons::parameters::{
+    ///     DecompositionBaseLog, DecompositionLevelCount, GlweDimension, PolynomialSize,
+    /// };
+    /// use concrete_core::prelude::*;
+    /// # use std::error::Error;
+    ///
+    /// # fn main() -> Result<(), Box<dyn Error>> {
+    /// // DISCLAIMER: the parameters used here are only for test purpose, and are not secure.
+    /// let glwe_dimension = GlweDimension(2);
+    /// let polynomial_size = PolynomialSize(4);
+    /// let level = DecompositionLevelCount(1);
+    /// let base_log = DecompositionBaseLog(4);
+    /// // Here a hard-set encoding is applied (shift by 20 bits)
+    /// let input = 3_u32 << 20;
+    /// let noise = Variance(2_f64.powf(-25.));
+    ///
+    /// let mut engine = CoreEngine::new()?;
+    /// let key: GlweSecretKey32 = engine.create_glwe_secret_key(glwe_dimension, polynomial_size)?;
+    /// let plaintext = engine.create_plaintext(&input)?;
+    ///
+    /// let ciphertext = engine.encrypt_ggsw_ciphertext(&key, &plaintext, noise, level, base_log)?;
+    /// #
+    /// assert_eq!(ciphertext.glwe_dimension(), glwe_dimension);
+    /// assert_eq!(ciphertext.polynomial_size(), polynomial_size);
+    ///
+    /// engine.destroy(key)?;
+    /// engine.destroy(plaintext)?;
+    /// engine.destroy(ciphertext)?;
+    /// #
+    /// # Ok(())
+    /// # }
+    /// ```
+    fn encrypt_ggsw_ciphertext(
+        &mut self,
+        key: &GlweSecretKey32,
+        input: &Plaintext32,
+        noise: Variance,
+        decomposition_level_count: DecompositionLevelCount,
+        decomposition_base_log: DecompositionBaseLog,
+    ) -> Result<GgswCiphertext32, GgswCiphertextEncryptionError<Self::EngineError>> {
+        Ok(unsafe {
+            self.encrypt_ggsw_ciphertext_unchecked(
+                key,
+                input,
+                noise,
+                decomposition_level_count,
+                decomposition_base_log,
+            )
+        })
+    }
+
+    unsafe fn encrypt_ggsw_ciphertext_unchecked(
+        &mut self,
+        key: &GlweSecretKey32,
+        input: &Plaintext32,
+        noise: Variance,
+        decomposition_level_count: DecompositionLevelCount,
+        decomposition_base_log: DecompositionBaseLog,
+    ) -> GgswCiphertext32 {
+        let mut ciphertext = ImplGgswCiphertext::allocate(
+            0u32,
+            key.polynomial_size(),
+            key.glwe_dimension().to_glwe_size(),
+            decomposition_level_count,
+            decomposition_base_log,
+        );
+        key.0.encrypt_constant_ggsw(
```

Comment:
Same question here (why is it constant?)

Reply:
See above.